microsoft/testsuites/vm_extensions/kvvm_extension.py

```diff
@@ -0,0 +1,211 @@
+from lisa.sut_orchestrator.azure.platform_ import AzurePlatform
+from lisa.testsuite import TestResult
+from lisa.util import SkippedException
+
+
+@TestSuiteMetadata(
+    area="vm_extension",
+    category="functional",
+    description="BVT for Azure Key Vault Extension",
+    requirement=simple_requirement(unsupported_os=[]),
+)
+class AzureKeyVaultExtensionBvt(TestSuite):
+    @TestCaseMetadata(
+        description="""
+        The following test case validates the Azure Key Vault Linux
+        Extension while creating the following resources:
+        A resource group
+        A VM
+        A Key Vault
+        Two certificates in the Key Vault
+        Retrieval of the certificate's secrets
+        through SecretClient class from Azure SDK.
+        Installation of the Azure Key Vault Linux Extension on the VM.
+        Rotation of the certificates (After KVVM Extension has been installed)
+        All of the resources have been created by using the Azure SDK Python.
+        """,
+        priority=1,
+        requirement=simple_requirement(supported_features=[AzureExtension]),
+    )
+    def verify_key_vault_extension(
+        self, log: Logger, node: Node, result: TestResult
+    ) -> None:
+        # Section for vault name and supported OS check
+        vault_name = os.getenv(
+            "vault_name", f"python-keyvault-{random.randint(1, 100000):05}"
+        )
+        vault_name_a = vault_name + "prp"
+        if isinstance(node.os, FreeBSD):
+            raise SkippedException(f"unsupported distro type: {type(node.os)}")
+
+        # Section for environment setup
+        environment = result.environment
+        assert environment, "fail to get environment from testresult"
+        platform = environment.platform
+        assert isinstance(platform, AzurePlatform)
+        # VM attributes
+        node_context = get_node_context(node)
+        resource_group_name = node_context.resource_group_name
+        node_capability = node.capability.get_extended_runbook(AzureNodeSchema, AZURE)
+        location = node_capability.location
+
+        # User's attributes
+        user_tenant_id = os.environ.get("tenant_id")
+        if user_tenant_id is None:
+            raise ValueError("Environment variable 'tenant_id' is not set.")
+        user_object_id = os.environ.get("user_object_id")
+        if user_tenant_id is None:
+            raise ValueError("Environment variable 'user_object_id' is not set.")
+        assert user_object_id is not None
+        credential = DefaultAzureCredential(additionally_allowed_tenants=["*"])
+
+        # Identity assignment
+        compute_client = ComputeManagementClient(credential, platform.subscription_id)
+        vm = compute_client.virtual_machines.get(
+            resource_group_name, node_context.vm_name
+        )
+        object_id_vm = vm.identity.principal_id
+        if object_id_vm is None:
+            raise ValueError("object_id_vm is not set.")
+
+        # Key Vault properties
```

Comment:
We could create wrappers in common.py for KeyVaultManagmentClient. For example, storage accounts are done similarly:

https://github.com/microsoft/lisa/blob/573570bd22f696b01f85bf3aabe98f838ddac1ef/lisa/sut_orchestrator/azure/common.py#L885

I have a WIP in this format, we can sync on it. Not blocking for your PR, I can make that improvement with my ADE PR

Reply:
Yep, there is a lot of room for improvement in the code, it was my first-time using LISA so definitely lots of part of the code that can be refactored.